cmd/root.go

```diff
@@ -53,14 +63,50 @@ func New() *cobra.Command {
 	})
 
 	flags := rootCmd.PersistentFlags()
-	flags.String("config", "", "Config file (default is $HOME/.hubble.yaml)")
-	flags.BoolP("debug", "D", false, "Enable debug messages")
-	flags.String("server", defaults.GetDefaultSocketPath(), "Address of a Hubble server")
-	flags.Duration("timeout", defaults.DefaultDialTimeout, "Hubble server dialing timeout")
+	flags.String("config", "", "Config file (default is $HOME/.hubble.yaml).")
+	flags.BoolP("debug", "D", false, "Enable debug messages.")
+	flags.String("server", defaults.GetDefaultSocketPath(), "Address of a Hubble server.")
+	flags.Duration("timeout", defaults.DefaultDialTimeout, "Hubble server dialing timeout.")
+	flags.Bool(
+		"tls",
+		false,
+		"Specify that TLS must be used when establishing a connection to a Hubble server.\r\n"+
+			"By default, TLS is only enabled if the server address starts with 'tls://'.",
+	)
+	flags.Bool(
+		"tls-allow-insecure",
+		false,
+		"Allows the client to skip verifying the server's certificate chain and host name.\r\n"+
+			"This option is NOT recommended as, in this mode, TLS is susceptible to machine-in-the-middle attacks.\r\n"+
+			"See also the 'tls-server-name' option which allows setting the server name.",
+	)
+	flags.StringSlice(
+		"tls-ca-cert-files",
+		nil,
+		"Paths to custom Certificate Authority (CA) certificate files."+
+			"The files must contain PEM encoded data.",
+	)
+	flags.String(
+		"tls-client-cert-file",
+		"",
+		"Path to the public key file for the client certificate to connect to a Hubble server (implies TLS).\r\n"+
+			"The file must contain PEM encoded data.",
+	)
+	flags.String(
+		"tls-client-key-file",
+		"",
+		"Path to the private key file for the client certificate to connect a Hubble server (implies TLS).\r\n"+
+			"The file must contain PEM encoded data.",
+	)
+	flags.String(
+		"tls-server-name",
+		"",
+		"Specify a server name to verify the hostname on the returned certificate (eg: 'instance.hubble-relay.cilium.io').",
+	)
 	vp.BindPFlags(flags)
 
 	rootCmd.SetErr(os.Stderr)
-	rootCmd.SetVersionTemplate("{{with .Name}}{{printf \"%s \" .}}{{end}}{{printf \"v%s\" .Version}}\n")
+	rootCmd.SetVersionTemplate("{{with .Name}}{{printf \"%s \" .}}{{end}}{{printf \"v%s\" .Version}}\r\n")
```

Comment:
What's the rationale for this? Windows support? I did not see any comment in the commit message.

Reply:
Oh right, just Windows support refinement :grin: